clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerResourceDefinition.java

```diff
@@ -0,0 +1,155 @@
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.clustering.infinispan.subsystem.remote;
+
+import java.util.function.UnaryOperator;
+
+import org.infinispan.client.hotrod.ProtocolVersion;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.jboss.as.clustering.controller.CapabilityProvider;
+import org.jboss.as.clustering.controller.CapabilityReference;
+import org.jboss.as.clustering.controller.ChildResourceDefinition;
+import org.jboss.as.clustering.controller.ManagementResourceRegistration;
+import org.jboss.as.clustering.controller.ResourceDescriptor;
+import org.jboss.as.clustering.controller.ResourceServiceBuilderFactory;
+import org.jboss.as.clustering.controller.ResourceServiceHandler;
+import org.jboss.as.clustering.controller.RestartParentResourceRegistration;
+import org.jboss.as.clustering.controller.UnaryRequirementCapability;
+import org.jboss.as.clustering.controller.validation.EnumValidator;
+import org.jboss.as.clustering.infinispan.subsystem.InfinispanExtension;
+import org.jboss.as.clustering.infinispan.subsystem.ThreadPoolResourceDefinition;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.registry.AttributeAccess;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.clustering.infinispan.spi.InfinispanRequirement;
+import org.wildfly.clustering.service.UnaryRequirement;
+
+/**
+ * /subsystem=infinispan/remote-cache-container=X
+ *
+ * @author Radoslav Husar
+ */
+public class RemoteCacheContainerResourceDefinition extends ChildResourceDefinition<ManagementResourceRegistration> {
+
+    public static final PathElement WILDCARD_PATH = pathElement(PathElement.WILDCARD_VALUE);
+
+    public static PathElement pathElement(String containerName) {
+        return PathElement.pathElement("remote-cache-container", containerName);
+    }
+
+    public enum Capability implements CapabilityProvider {
+        CONTAINER(InfinispanRequirement.REMOTE_CONTAINER),
+        CONFIGURATION(InfinispanRequirement.REMOTE_CONTAINER_CONFIGURATION),
+        ;
+
+        private final org.jboss.as.clustering.controller.Capability capability;
+
+        Capability(UnaryRequirement requirement) {
+            this.capability = new UnaryRequirementCapability(requirement);
+        }
+
+        @Override
+        public org.jboss.as.clustering.controller.Capability getCapability() {
+            return this.capability;
+        }
+    }
+
+    public enum Attribute implements org.jboss.as.clustering.controller.Attribute {
+        CONNECTION_TIMEOUT("connection-timeout", ModelType.INT, new ModelNode(60000)),
```

Comment:
We need to add a "module" attribute used to configure the marshaller.

Reply:
Done; added new commit to make review easier.